python/lsst/ip/isr/photodiode.py

```diff
@@ -0,0 +1,260 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+Photodiode storage class.
+"""
+import numpy as np
+from astropy.table import Table
+
+from lsst.ip.isr import IsrCalib
+
+
+__all__ = ["PhotodiodeCalib"]
+
+
+class PhotodiodeCalib(IsrCalib):
+    """Independent current measurements from photodiode for linearity
+    calculations.
+
+    Parameters
+    ----------
+    timeSamples : `list` or `numpy.ndarray`
+        List of samples the photodiode was measured at.
+    currentSamples : `list` or `numpy.ndarray`
+        List of current measurements at each time sample.
+    log : `lsst.log.Log`, optional
+        Log to write messages to.
+    **kwargs :
+        Paramters to pass to parent constructor.
+    """
+
+    _OBSTYPE = 'PHOTODIODE'
+    _SCHEMA = 'Gen3 Photodiode'
+    _VERSION = 1.0
+
+    def __init__(self, timeSamples=None, currentSamples=None, **kwargs):
+        if timeSamples is not None and currentSamples is not None:
+            if len(timeSamples) != len(currentSamples):
+                raise RuntimeError(f"Inconsitent vector lengths: {len(timeSamples)} {len(currentSamples)}")
+            else:
+                self.timeSamples = np.array(timeSamples)
+                self.currentSamples = np.array(currentSamples)
+        else:
+            self.timeSamples = np.array([])
+            self.currentSamples = np.array([])
+
+        super().__init__(**kwargs)
+
+        if 'integrationMethod' in kwargs:
+            self.integrationMethod = kwargs.pop('integrationMethod')
+        else:
+            self.integrationMethod = 'DIRECT_SUM'
+
+        if 'day_obs' in kwargs:
+            self.updateMetadata(day_obs=kwargs['day_obs'])
+        if 'seq_num' in kwargs:
+            self.updateMetadata(seq_num=kwargs['seq_num'])
+
+        self.requiredAttributes.update(['timeSamples', 'currentSamples', 'integrationMethod'])
+
+    @classmethod
+    def fromDict(cls, dictionary):
+        """Construct a PhotodiodeCalib from a dictionary of properties.
+
+        Parameters
+        ----------
+        dictionary : `dict`
+            Dictionary of properties.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.PhotodiodeCalib`
+            Constructed photodiode data.
+
+        Raises
+        ------
+            Raised if the supplied dictionary is for a different
+            calibration type.
+        """
+        calib = cls()
+
+        if calib._OBSTYPE != dictionary['metadata']['OBSTYPE']:
+            raise RuntimeError(f"Incorrect photodiode supplied.  Expected {calib._OBSTYPE}, "
+                               f"found {dictionary['metadata']['OBSTYPE']}")
+
+        calib.setMetadata(dictionary['metadata'])
+
+        calib.timeSamples = np.array(dictionary.get('timeSamples'))
+        calib.currentSamples = np.array(dictionary.get('currentSamples'))
+        calib.integrationMethod = dictionary.get('integrationMethod')
```

Comment:
Do you really want to use `.get()` here rather than just a direct `['timeSamples']`? I would have thought that they're mandatory, and so the KeyError there would be a feature? Or do you want the `None`s back for some reason?